### PR TITLE
TOK-136: Add integration tests for skipping distribution

### PR DESCRIPTION
**Pull request by nmagariang**
## What

Add integration tests to cover scenarios where distribution is skipped

## Refs

- [Task](https://rsklabs.atlassian.net/browse/TOK-136)

### CODE REVIEW

File: test/integration/SkipDistribution.t.sol [hunk 1]
```diff
@@ -0,0 +1,969 @@
...
+        //  WHEN alice votes to gauge and gauge2
+        vm.startPrank(alice);
+        sponsorsManager.allocate(gauge, 1 ether);
+        sponsorsManager.allocate(gauge2, 1 ether);
+        vm.stopPrank();
+        // AND bob votes to gauge
+        vm.prank(bob);
+        sponsorsManager.allocate(gauge, 2 ether);
+
+        // EPOCH 2
```
Comment (jmendiola222) [inline test/integration/SkipDistribution.t.sol:36]: how did we go from EPOCH 1 to 2 here? I don't see nothing moving time, just Alice and Bob allocations.

Comment (nmagariang) [reply test/integration/SkipDistribution.t.sol:36]: It happens in the [_distribute()](https://github.com/RootstockCollective/collective-rewards-sc/blob/7accaf0b47a50d53f17b54159072a8966064b12f/test/BaseTest.sol#L170) method, maybe the change of epoch should be written after adding a clarification to what the distribute does?

File: test/integration/SkipDistribution.t.sol [hunk 2]
```diff
@@ -0,0 +1,969 @@
...
+        sponsorsManager.allocate(gauge2, 1 ether);
+
+        // EPOCH 4
+        // AND 100 rewardTokens and 10 coinbase are distributed in next epoch
+        _distribute(100 ether, 10 ether);
+
+        // AND epoch finishes
+        _skipAndStartNewEpoch();
+
+        // EPOCH 4
```
Comment (nmagariang) [inline test/integration/SkipDistribution.t.sol:56]: This one's 5

**Review by jmendiola222 — APPROVED** [review]
LGTM, small future suggestion: having the normal case along the "deviation" is useful to compare them.

**Review by franciscotobar — APPROVED** [review]
LGTM

**Pull request merged**